File: Python/solucion-2018/sabor.py
class Sabor:
	def __init__(self, nombre="", hidratos=0.0, proteinas=0.0, grasas=0.0):
		self.__nombre = nombre
		self.__hidratos = hidratos
		self.__proteinas = proteinas
		self.__grasas = grasas
		
	def __str__(self):
		return f"Sabor {self.__nombre} ({self.__hidratos}, {self.__proteinas}, {self.__grasas})"
		
	def get_nombre(self):
		return self.__nombre
	def get_hidratos(self):
		return self.__hidratos
	def get_proteinas(self):
		return self.__proteinas
	def get_grasas(self):
		return self.__grasas
		
	def set_nombre(self, nombre):
		self.__nombre = nombre
	def set_hidratos(self, hidratos):
		self.__hidratos = hidratos
	def set_proteinas(self, proteinas):
		self.__proteinas = proteinas
	def set_grasas(self, grasas):
		self.__grasas = grasas

File: Java/DeustoGelati/Sabor.java
package POO;

public class Sabor {
	
	//Atributos
	private String nombre;
	private float hidratos;
	private float proteinas;
	private float grasas;
	
	//Getters y setters
	public String getNombre() {
		return this.nombre;
	}
	public void setNombre(String nombre) {
		this.nombre = nombre;
	}
	public float getHidratos() {
		return this.hidratos;
	}
	public void setHidratos(float hidratos) {
		this.hidratos = hidratos;
	}
	public float getProteinas() {
		return this.proteinas;
	}
	public void setProteinas(float proteinas) {
		this.proteinas = proteinas;
	}
	public float getGrasas() {
		return this.grasas;
	}
	public void setGrasas(float grasas) {
		this.grasas = grasas;
	}
	
	//Constructores
	public Sabor() {
		this.nombre = "Chocolate";
		this.hidratos = 0;
		this.proteinas = 0;
		this.grasas = 0;
	}
	public Sabor(String nombre, float hidratos, float proteinas, float grasas) {
		this.nombre = nombre;
		this.hidratos = hidratos;
		this.proteinas = proteinas;
		this.grasas = grasas;
	}
	public Sabor(Sabor sabor) {
		this.nombre = sabor.nombre;
		this.hidratos = sabor.hidratos;
		this.proteinas = sabor.proteinas;
		this.grasas = sabor.grasas;
	}
	
	//toString()
	public String toString() {
		return this.nombre + " (hidratos: " + this.hidratos +
			", proteinas: " + this.proteinas + ", grasas: " + this.grasas + ")";
	}

}
